tensorflow_probability/python/distributions/sigmoid_beta.py

```diff
@@ -0,0 +1,278 @@
+                                                 eps(dtype)))))
+    # pylint: enable=g-long-lambda
+
+  @property
+  def concentration1(self):
+    """Concentration parameter associated with a `1` outcome."""
+    return self._concentration1
+
+  @property
+  def concentration0(self):
+    """Concentration parameter associated with a `0` outcome."""
+    return self._concentration0
+
+  def _default_event_space_bijector(self):
+    return identity_bijector.Identity(validate_args=self.validate_args)
+
+  def _batch_shape_tensor(self, concentration1=None, concentration0=None):
+    return ps.broadcast_shape(
+          ps.shape(self.concentration1
+                   if concentration1 is None else concentration1),
+          ps.shape(self.concentration0
+                   if concentration0 is None else concentration0))
+
+  def _batch_shape(self):
+    return tf.broadcast_static_shape(self.concentration1.shape,
+                                     self.concentration0.shape)
+
+  def _parameter_control_dependencies(self, is_init):
+    if not self.validate_args:
+      return []
+    assertions = []
+    for concentration in [self.concentration0, self.concentration1]:
+      if is_init != tensor_util.is_ref(concentration):
+        assertions.append(
+          assert_util.assert_positive(
+            concentration,
+            message='Concentration parameter must be positive.'))
+    return assertions
+
+  def _event_shape_tensor(self):
+    return tf.constant([], dtype=tf.int32)
+
+  def _event_shape(self):
+    return tf.TensorShape([])
+
+  def _sample_n(self, n, seed=None):
+    seed1, seed2 = samplers.split_seed(seed, salt='sigmoid_beta')
+    concentration1 = tf.convert_to_tensor(self.concentration1)
+    concentration0 = tf.convert_to_tensor(self.concentration0)
+    shape = self._batch_shape_tensor(concentration1, concentration0)
+    expanded_concentration1 = tf.broadcast_to(concentration1, shape)
+    expanded_concentration0 = tf.broadcast_to(concentration0, shape)
+    log_gamma1 = gamma_lib.random_gamma(
+        shape=[n],
+        concentration=expanded_concentration1,
+        seed=seed1,
+        log_space=True)
+    log_gamma2 = gamma_lib.random_gamma(
+        shape=[n],
+        concentration=expanded_concentration0,
+        seed=seed2,
+        log_space=True)
+
+    return log_gamma1 - log_gamma2
+
+  def _log_normalization(self, concentration0, concentration1):
+    return tfp_math.lbeta(concentration0, concentration1)
+
+  def _log_unnormalized_prob(self, concentration0, concentration1, x):
+    out = -(concentration0 + concentration1) * tf.math.log1p(
+        tf.math.exp(-x))
+    return out - concentration1 * x
+
+  def _log_prob(self, x):
+    a = tf.convert_to_tensor(self.concentration1)
+    b = tf.convert_to_tensor(self.concentration0)
+    return self._log_unnormalized_prob(a, b, x) - \
+        self._log_normalization(a, b)
+
+  def _log_cdf(self, x):
```

Comment:
I think this should be implemented by default in the base class when _cdf is implemented